docs/DeveloperGuide.md

```diff
@@ -1,42 +1,177 @@
 6. `ViewMangasCommand`
-7. `GreetCommand`
-8. `ByeCommand`
-#### Command Structure
+7. `AddSalesCommand`
+8. `GreetCommand`
+9. `ByeCommand`
 
+#### Command Structure
 All child `Command` classes must inherit from the abstract `Command` class. Each child class is required to implement the abstract `execute` method.
 
 While child classes may or may not modify the `AuthorList`, they are encouraged to utilize the `Ui` class to interact with users, such as displaying success messages.
 
 #### Guidelines for Future Developers
-
 When adding new command classes, developers must follow the same method of implementation by 
 inheriting from the abstract `Command` class. Ensure that each new command class includes 
 an implementation of the `execute` method and appropriately interacts with the `Ui` class 
 for user feedback. Additionally, developers should update the `Parser` class to gather the 
 relevant arguments from the user for their commands.
 
+### Saving Data
+`AuthorList` data accumulated by the user can be saved with the provided `Storage` and `StorageHelper` classes. 
+
+#### Storage Structure
+![StorageClass.png](uml/puml/StorageClass/StorageClass.png)
+The above UML class diagram outlines the structure of the classes related to saving data. 
+
+The `Storage` class uses the Singleton design pattern, which means only a maximum of one `Storage` instance can exist during the program's lifespan. To access it, call the static method `Storage::getInstance`. 
+
+The `StorageHelper` utility class wraps the methods to access `Storage` for ease of use.
+
+Data is by default stored in a JSON file `catalog.json` in the `data` directory at the program root location. This can be changed via the `public static final String DATA_PATH` constant in the `Storage.java` file. 
+
+The class makes use of the `Gson` third-party library to de/serialize data. 
+
+When needed, call `StorageHelper::readFile` to return the deserialized `AuthorList` from `catalog.json`. 
+
+Whenever a user action that modifies the state of the `AuthorList` is performed, the corresponding overridden `Command::execute` method should call `StorageHelper.saveFile(authorList: AuthorList)` after modifying the data.     
+
+#### Storage Behaviour
+The following UML sequence diagrams outline the behaviour of the program when the user inputs a command that modifies the `AuthorList`. 
+![SavingDataSequence.png](uml/puml/SavingDataSequence/SavingDataSequence.png)
+![refGetStorageInstanceSequence.png](uml/puml/SavingDataSequence/refGetStorageInstanceSequence.png)
+
+#### Gson De/serialization
+Instead of using the default deserializers provided by `Gson`, this project defines custom ones. This enables us to perform checks on the key-value pairs in the data file every step of the way, providing detailed and relevant information in the event deserialization is not successful. The following is a code snippet showcasing the checks performed during the deserialization of data. 
+```
```

Comment:
Good job explaining the Gson deserialization and serialization! The functions are actually quite complex, thus this will be useful for future developers of this project.